apis/kueue/webhooks/clusterqueue_webhook.go

```diff
@@ -158,7 +158,11 @@ func validateFlavorQuotas(flavorQuotas kueue.FlavorQuotas, coveredResources []co
 		allErrs = append(allErrs, field.Invalid(path.Child("resources"), field.OmitValueType{}, "must have the same number of resources as the coveredResources"))
 	}
 
+	coveredResNum := len(coveredResources)
 	for i, rq := range flavorQuotas.Resources {
+		if coveredResNum < i+1 {
```

Comment:
nit (just my personal taste)

```suggestion
		if i >= len(coveredResNum) {
```

I wouldn't imagine that `len` is costly. slices are basically a struct of one pointer and two numbers (len and cap).